### PR TITLE
Improve unit test coverage

### LINKED ISSUE #86: add unit tests to increase code coverage to over 80%

### PR DESCRIPTION
**Pull request by joao-borges**
Fixes #86

### CODE REVIEW

**Comment by joao-borges** [comment]
This is not ready yet - I just want to see the sonar reports.

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleDocumentBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.builder;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BundleDocumentBuilderTest {
```
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleDocumentBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eprChX7_AlPFEV&open=AWajX2eprChX7_AlPFEV)

[ISSUE_KEY]: AWajX2eprChX7_AlPFEV

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/ListenPortEntityBuilder.java [hunk 2]
```diff
@@ -85,7 +85,7 @@
             "TLS_RSA_WITH_AES_128_GCM_SHA256",
             "TLS_RSA_WITH_AES_128_CBC_SHA256",
             "TLS_RSA_WITH_AES_128_CBC_SHA"));
-    static final List<String> TLS_VERSIONS = asList(TLSV1, TLSV11, TLSV12);
+    public static final List<String> TLS_VERSIONS = asList(TLSV1, TLSV11, TLSV12);
```
Comment (apigatewayopensource) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/ListenPortEntityBuilder.java]: Vulnerability Vulnerability: Make this member "protected". ([squid:S2386](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2386))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2c-rChX7_AlPFEP&open=AWajX2c-rChX7_AlPFEP)

[ISSUE_KEY]: AWajX2c-rChX7_AlPFEP

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleEntityBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.builder;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BundleEntityBuilderTest {
```
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleEntityBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eirChX7_AlPFEU&open=AWajX2eirChX7_AlPFEU)

[ISSUE_KEY]: AWajX2eirChX7_AlPFEU

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/PolicyEntityBuilderTest.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.builder;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PolicyEntityBuilderTest {
```
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/PolicyEntityBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2esrChX7_AlPFEW&open=AWajX2esrChX7_AlPFEW)

[ISSUE_KEY]: AWajX2esrChX7_AlPFEW

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/EncassLoaderTest.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.bundle.loader;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class EncassLoaderTest {
```
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/EncassLoaderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eRrChX7_AlPFET&open=AWajX2eRrChX7_AlPFET)

[ISSUE_KEY]: AWajX2eRrChX7_AlPFET

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/FolderLoaderTest.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.bundle.loader;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FolderLoaderTest {
```
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/FolderLoaderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2d_rChX7_AlPFEQ&open=AWajX2d_rChX7_AlPFEQ)

[ISSUE_KEY]: AWajX2d_rChX7_AlPFEQ

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/IdentityProviderLoaderTest.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.bundle.loader;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class IdentityProviderLoaderTest {
```
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/IdentityProviderLoaderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eBrChX7_AlPFER&open=AWajX2eBrChX7_AlPFER)

[ISSUE_KEY]: AWajX2eBrChX7_AlPFER

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/PolicyLoaderTest.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.bundle.loader;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PolicyLoaderTest {
```
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/PolicyLoaderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eOrChX7_AlPFES&open=AWajX2eOrChX7_AlPFES)

[ISSUE_KEY]: AWajX2eOrChX7_AlPFES

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleDocumentBuilderTest.java (hunk 1 above)
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleDocumentBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eprChX7_AlPFEV&open=AWajX2eprChX7_AlPFEV)

[ISSUE_KEY]: AWajX2eprChX7_AlPFEV

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/PolicyEntityBuilderTest.java (hunk 4 above)
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/PolicyEntityBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2esrChX7_AlPFEW&open=AWajX2esrChX7_AlPFEW)

[ISSUE_KEY]: AWajX2esrChX7_AlPFEW

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleEntityBuilderTest.java (hunk 3 above)
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleEntityBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eirChX7_AlPFEU&open=AWajX2eirChX7_AlPFEU)

[ISSUE_KEY]: AWajX2eirChX7_AlPFEU

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/PolicyEntityBuilderTest.java (hunk 4 above)
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/PolicyEntityBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2esrChX7_AlPFEW&open=AWajX2esrChX7_AlPFEW)

[ISSUE_KEY]: AWajX2esrChX7_AlPFEW

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleEntityBuilderTest.java (hunk 3 above)
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleEntityBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eirChX7_AlPFEU&open=AWajX2eirChX7_AlPFEU)

[ISSUE_KEY]: AWajX2eirChX7_AlPFEU

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleDocumentBuilderTest.java (hunk 1 above)
Comment (apigatewayopensource) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/BundleDocumentBuilderTest.java]: Code Smell Code Smell: Add some tests to this class. ([squid:S2187](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2187))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=98&issues=AWajX2eprChX7_AlPFEV&open=AWajX2eprChX7_AlPFEV)

[ISSUE_KEY]: AWajX2eprChX7_AlPFEV

**Pull request merged**